src/System.ServiceProcess.ServiceController/tests/System.ServiceProcess.ServiceController.TestService/TestService.cs

```diff
@@ -38,7 +38,12 @@ public TestService(string serviceName, Exception throwException = null)
             this._exception = throwException;
 
             this._serverStream = new NamedPipeServerStream(serviceName);
-            _waitClientConnect = this._serverStream.WaitForConnectionAsync();
+            this._serverStream.BeginWaitForConnection( new AsyncCallback(SendConnectedByte), this);
```

Comment:
IAsyncResult is the 'legacy' async mechasism. The preferred way is to now use async/await and Tasks, ie use WaitForConnectionAsync  and then have it ".ContinueWith()" with the desired next step. Try that.